splunk/src/main/java/com/splunk/Service.java

```diff
@@ -1319,7 +1325,17 @@ public Job search(String query, Map<String, Object> args) {
         if (token != null && !cookieStore.hasSplunkAuthCookie() ) {
             request.getHeader().put("Authorization", token);
         }
-        return super.send(fullpath(path), request);
+        ResponseMessage responseMessage = super.send(fullpath(path), request);
+        if(responseMessage.getStatus() == 401 && (this.autologin && this.username!= null && this.password != null)){
+            // Executing re-login to renew the session token.
+            System.out.println("Re-login executed");
+            this.login(this.username, this.password);
+            responseMessage = super.send(fullpath(path), request);
+        }else if(responseMessage.getStatus() >= 400){
+            //if autologin is not set to true or username/password is not set, throw HTTPException

```

Comment:
I think the else is also meant to handle every other error code that is not 401, could we update the comment to reflect that? ->  "if not handling 401 and autologin scenario"